### PR TITLE
Update TES executor to TES API v1.1

### PR DESCRIPTION
**Pull request by bentsherman**
Summary of changes:
- Update TES Java SDK to TES 1.1 spec
- Deprecate `NXF_EXECUTOR_TES_ENDPOINT` in favor of `tes.endpoint` config option
- Add support for output globs
- Add support for output directories
- Update docs
- Add README with instructions for Java SDK generation

Unfortunately Funnel hasn't been updated to 1.1 yet, so I don't have an easy way to test it at the moment.

@vsmalladi can you test this PR with Azure?

```bash
git clone https://github.com/nextflow-io/nextflow -b tes-update-1.1
cd nextflow
make compile

./launch.sh run ...
```

### CODE REVIEW

**Comment by bentsherman** [comment]
Release candidate for Funnel with TES 1.1 support: https://github.com/ohsu-comp-bio/funnel/releases/tag/0.11.0-rc.2

**Comment by lbeckman314** [comment]
Hi @bentsherman! Just wanted to reach out and check if there were any updates or blockers with this PR that we can help with. Let us know and we'll happy to take a look!

**Comment by bentsherman** [comment]
We're just trying to get TES Azure to work and then I think we'll merge

**Comment by vsmalladi** [comment]
Have an outstanding PR that is almost done. Should be able to finish this week.

File: plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy [hunk 1]
```diff
@@ -171,26 +172,30 @@ class TesTaskHandler extends TaskHandler {
...
-        // the cmd list to launch it
-        def job = new ArrayList(BashWrapperBuilder.BASH) << wrapperFile.getName()
-        List cmd = ['/bin/bash','-c', job.join(' ') + " &> $TaskRun.CMD_LOG" ]
+        if( !task.container )
+            throw new ProcessUnrecoverableException("Process `${task.lazyName()}` failed because the container image was not specified")
 
-        def exec = new TesExecutorModel()
-        exec.command = cmd
+        final exec = new TesExecutorModel()
+        exec.command = List.of('/bin/bash', '-c', "${BashWrapperBuilder.BASH} ${wrapperFile.getName()} &> ${TaskRun.CMD_LOG}".toString())
```
Comment (lbeckman314) [inline plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy:179]: @vsmalladi here is the change made that may require a small change in how Task Commands are handled by the executor:

Old Task Command:
```sh
/bin/bash -c /bin/bash -ue .command.run &> .command.log
```

New Task Command (with a nested List):
```sh
/bin/bash -c [/bin/bash -ue] .command.run &> .command.log
```

Comment (vsmalladi) [reply plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy:179]: Tahnks I think i got that in the Azure PR as well.

Comment (vsmalladi) [reply plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy:179]: https://github.com/nextflow-io/nextflow/blob/5e9ea644ff34c49ae584b3bbfeca51ef55a76e97/plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy#L194

I have it now for Azure. Should we just make it for all TES ?

**Comment by vsmalladi** [comment]
@bentsherman Whats the next steps to get this out of draft?

**Comment by bentsherman** [comment]
@vsmalladi I just looked over everything in total and it looks good, you should probably do another test with TES Azure to make sure it still works with my last round of edits

**Comment by vsmalladi** [comment]
@bentsherman built and hit this error:

`ERROR ~ Unexpected error [ProviderMismatchException]`

Thoughts?

**Comment by bentsherman** [comment]
Need more context that just the error message. How you ran it, the stack trace, etc

File: plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesExecutor.groovy [hunk 2]
```diff
@@ -74,23 +83,74 @@ class TesExecutor extends Executor implements ExtensionPoint {
...
-            def tempBin = getTempDir()
+            final tempBin = getTempDir()
             log.info "Uploading local `bin` scripts folder to ${tempBin.toUriString()}/bin"
             remoteBinDir = FilesEx.copyTo(session.binDir, tempBin)
+
+            remoteBinFiles = []
+            session.binDir.eachFileRecurse { file ->
+                if( file.isDirectory() )
+                    return
+                remoteBinFiles << tempBin.resolve(session.binDir.relativize(file).toString())
```
Comment (vsmalladi) [inline plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesExecutor.groovy:104]: @bentsherman 

This is not getting the correct url of the file path to the tes task.

`nftes3dd92310def6b/work/tmp/c5/993adc8e95593d80e56469aef20d78/run.sh`
but should be
`nftes3dd92310def6b/work/tmp/c5/993adc8e95593d80e56469aef20d78/bin/run.sh`

**Comment by vsmalladi** [comment]
@bentsherman test pass. I am able to submit to azure now. using this build.

I think we can merge this in.

**Comment by lbeckman314** [comment]
Tests passing with latest Nextflow PR and Funnel. Good to merge! :white_check_mark:

**Review by pditommaso — APPROVED** [review]

**Pull request merged**